src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -9597,6 +9416,12 @@ private ExpressionSyntax ParseTerm(Precedence precedence)
                         {
                             expr = this.ParseLambdaExpression();
                         }
+                        else if (this.IsPossibleDeconstructionLeft(precedence))
+                        {
+                            TypeSyntax type = this.ParseType();
```

Comment:
>TypeSyntax type = this.ParseType(); [](start = 28, length = 35)

It looks like the code in this block is a dupe of a new code in ParseSubExpressionCore. Consider extracting into a method and reusing.